src/io/flutter/logging/FlutterLogTree.java

```diff
@@ -281,27 +291,36 @@ public boolean accept(@NotNull FlutterLogEntry entry) {
       if (text == null) {
         return true;
       }
-
-      if (entry.getMessage().contains(text)) {
+      final String standartText = isMatchCase ? text : text.toLowerCase();
+      final String standartMessage = isMatchCase ? entry.getMessage() : entry.getMessage().toLowerCase();
+      final String standartCategory = isMatchCase ? entry.getCategory() : entry.getCategory().toLowerCase();
+      if (acceptByCheckingRegexOption(standartMessage, standartText)) {
         return true;
       }
+      return acceptByCheckingRegexOption(standartCategory, standartText);
+    }
 
-      //noinspection RedundantIfStatement
-      if (entry.getCategory().contains(text)) {
-        return true;
+    private boolean acceptByCheckingRegexOption(@NotNull String message, @NotNull String text) {
+      final String standardString = message.replaceAll("\n", "");
```

Comment:
I'm curious, why is this necessary?

Reply:
Because java regex doesn't recognize as a character.
For example:

```java
public class Main {
    public static void main(String args[]) {
        String input = "\n hello world \n";
        String regex = ".*hello.*";
        System.out.println("Result: " + input.matches(regex));
    }
}
```
Result: `false`.

But if we change input to:
```
String input = "\n hello world \n".replaceAll("\n", "");
```
=> Result: `true`.

Reply:
The affect of this code is user can't filter with string context break line.

Reply:
Aha.  Thanks!  If this is just for the regex case, should we move the `replaceAll` into the if block?

IOW, something like:
```java
if (isRegex) {
  // Extra replace to support matching on strings w/ line breaks.
  return message.replaceAll("\n", "").matches(".*" + text + ".*");
}
```

Reply:
Yeap. I updated. Thank you!

Reply:
Cool!  One last question...  I'm rusty w/ regexps but I wonder if we there isn't some combination of Pattern flags we couldn't use to get around need to modify the message string.  Maybe look at [`DOTALL`](https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html#DOTALL) and [`MULTILINE`](https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html#MULTILINE)?

Happy to land an iterate as well.

Thanks again!

Reply:
Yes, It looks better.
We just need to add `(?s)` to support DOTALL.
```
return message.matches("(?s).*" + text + ".*");
```
That's great!.
Thank you!